uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -201,6 +201,11 @@ async def run_asgi(self):
         """
         try:
             result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
+        except (concurrent.futures.CancelledError, asyncio.CancelledError):
+            self.logger.error("ASGI callable was cancelled while running")
+            if not self.handshake_started_event.is_set():
+                self.send_500_response()
+            self.transport.close()
```

Comment:
This block is not needed. This will be part of the `BaseException` block anyway, just with the fact that it doesn't wait the handshake to complete...